### PR TITLE
Add support for user/password authentication

### PR DESCRIPTION
**Pull request by rmohr**
Hi,

@rgbkrk  great project.

I added basic user/password authentication to libvirt-go.
The PR is not finished yet but it already works. 

It would be gread if I could get some feedback.

### CODE REVIEW

File: libvirt.go [hunk 1]
```diff
@@ -149,6 +181,23 @@ func (c *VirConnection) UnregisterCloseCallback() error {
 	return nil
 }
 
+//export authCallback
```
Comment (rgbkrk) [inline libvirt.go]: What do you mean by export here? auth is not capitalized here.

File: libvirt.go [hunk 2]
```diff
@@ -80,6 +83,34 @@ func NewVirConnection(uri string) (VirConnection, error) {
 	return obj, nil
 }
 
+func NewVirConnectionWithAuth(uri string, username string, password string) (VirConnection, error) {
```
Comment (rgbkrk) [inline libvirt.go]: Interface to the function matches the C function. :+1:

**Comment by rgbkrk** [comment]
Would you be able to right a test for this?

**Comment by rgbkrk** [comment]
Thanks for the contribution, seems to be aligned pretty well.

**Comment by rmohr** [comment]
Rewrote it to comply with cgo 1.6 pointer rules. Thanks for the initial review. 

I will add a test too.

**Comment by vincentbernat** [comment]
I think you could wrap everything in C code. This way, you just have to invoke the C function from Go code and not handle all the callback and allocation stuff in Go code.

**Comment by rmohr** [comment]
> I think you could wrap everything in C code. This way, you just have to invoke the C function from Go code and not handle all the callback and allocation stuff in Go code.

I'll try it, but I don't think it would save me more than one or two C.free calls. Let's see..

**Comment by rmohr** [comment]
> Would you be able to right a test for this?

Done

File: libvirt.go (hunk 2 above)
Comment (rmohr) [inline libvirt.go]: SASL has more than just these two mechs.
Maybe I should be more specific in the method name that this is about username and password authentication? 
Of course we could work here with callbacks too ... Thoughts?

**Comment by rgbkrk** [comment]
Thanks for the contribution @rmohr, looking forward to having you around here.

**Pull request merged**